src/osd/ScrubResult.cc

```diff
@@ -0,0 +1,224 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+
+#include "ScrubResult.h"
+#include "osd_types.h"
+#include "include/rados/rados_types.hpp"
+
+namespace Scrub {
+
+void ShardError::on_shard_missing()
```

Comment:
what about **set_shard_missing()** instead ? The **on_** makes me think there is a functional side effect that does not modify the object.